pkg/zabbix_api.go

```diff
@@ -429,14 +440,20 @@ func (ds *ZabbixDatasource) getHosts(ctx context.Context, dsInfo *datasource.Dat
 	if err != nil {
 		return nil, err
 	}
+	isRegex, hostFilter := parseFilter(hostFilter)
+	re := regexp.MustCompile(hostFilter)
```

Comment:
Don't use `MustCompile` if the filter isn't meant to be a regex. If it's not a valid pattern, it will panic.

Also perhaps we want to use something that returns an error when it fails to compile so we can surface that to the user as a user error.

Reply:
Updated, 04b1b2a

Reply:
b6c2116
Cleaned it up a bit and added unit tests for parseFilter